controllers/vspherecluster_controller.go

```diff
@@ -282,6 +270,13 @@ func (r clusterReconciler) reconcileNormal(ctx *context.ClusterContext) (reconci
 		return reconcile.Result{}, nil
 	}
 
+	// If the cluster is deleted, that's mean that the workload cluster is being deleted and so the CCM/CSI instances
+	if !ctx.Cluster.DeletionTimestamp.IsZero() {
+		conditions.MarkFalse(ctx.VSphereCluster, infrav1.CCMAvailableCondition, clusterv1.DeletingReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
It looks like this block would only get exercised if the cluster is deleted while in the middle of a normal reconcile. Otherwise it would go through `reconcileDelete`.  Should this be part of reconcileDelete as well?

Reply:
Done!